src/utils/utils_posix_common.c

```diff
@@ -266,3 +267,26 @@ int utils_file_open_or_create(const char *path) {
 
     return fd;
 }
+
+// Expected input:
+// char *str_threshold = utils_env_var_get_str("UMF_PROXY", "size.threshold=");
+long utils_get_size_threshold(char *str_threshold) {
+    if (!str_threshold) {
+        return 0;
+    }
+
+    // move to the beginning of the number
+    str_threshold += strlen("size.threshold=");
+
+    // check if the first character is a digit
+    if (!isdigit(str_threshold[0])) {
+        LOG_ERR("incorrect size threshold: %s", str_threshold);
+        return -1;
+    }
+
+    size_t threshold = (size_t)atol(str_threshold);
```

Comment:
from `atol` docs:

`If the converted value would be out of the range of representable values by a long int, it causes undefined behavior.` - can we check it somehow easily...? after all, this is an input from user

Reply:
How do you want to check that?